frontend/src/views/DashboardPage/components/MoveSecrets/MoveSecretsToFolder.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react"
+import { twMerge } from "tailwind-merge";
+
+import {
+  Button, ModalClose,
+} from "@app/components/v2";
+
+import { TSecretFolder } from "../../../../hooks/api/secretFolders/types";
+import {DirectorySelector} from "./DirectorySelector";
+
+type Props = {
+  checkedSecrets: { _id: string, isChecked: string | boolean }[],
+  onMoveSecrets: (folderId: string, checkedSecrets: { _id: string, isChecked: string | boolean }[]) => void;
+  folderData: {
+    dir: TSecretFolder[];
+    folders: TSecretFolder[];
+  } | undefined
+};
+
+export const MoveSecretsToFolder = ({ folderData, checkedSecrets, onMoveSecrets }: Props): JSX.Element => {
+  const directoryData = folderData
+  const [selectedPath, setSelecctedPath] = useState<string>("")
+  const [folderId, setFolderId] = useState<string>("")
+
+  const handleSelectPath = (chosenPath: string, folderID: string) => {
+    setSelecctedPath(chosenPath)
```

Comment:
Typo

Reply:
Concerning the props optimisation here. So I am adding a useCallback to memoize the handleCheckedSecret function

Reply:
I have also addressed the optimisation concerns for the new props.